docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,686 @@
+    * **type**: defines what kind of plugin to use, it's a string type so any new plugins should work.
+    * **config**: json map that is mapped to configuration provided in the type.
+  * **endpoints**: defines a list of endpoints.
+    * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+    * **port**: defines a port of a destination.
+  * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
+    * **enabled**: defines if proxy should originate TLS.
+    * **version**: section for providing version specification.
+      * **min**: defines minmum supported version. One of `TLSAuto`, `TLS10`, `TLS11`, `TLS12`, `TLS13`
+      * **max**: defines maximum supported version. One of `TLSAuto`, `TLS10`, `TLS11`, `TLS12`, `TLS13`
+    * **allowRenegotiation**: defines if TLS sessions will allow renegotiation.
+    * **verification**: section for providing TLS verification details.
+      * **skipVerification**: defines if proxy should skip verification, one of `SkipSAN`, `SkipCA`, `Secured`, `SkipALL`. Default `Secured`.
+      * **subjectAltNames**: list of names to verify in the certificate.
+      * **caCert**: defines a certificate of CA.
+        * one of `inline`, `inlineString` or `secret`.
+      * **clientCert**: defines a certificate of a client.
+        * one of `inline`, `inlineString` or `secret`.
+      * **clientKey**: defines a client private key.
+        * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **value**: allocated IP for a provided domain with `KumaHostname` type in a match section or provided IP
+    * **type**: provides information about the way IP was provided
+  * **addresses**: section for generated domain
+    * **hostname**: generated domain 
+    * **status**: indicate if an address is available
+    * **origin**: section providing information what generated the vip
+      * **kind**: points to kind that generated domain
+      * **name**: name of the `HostnameGenerator`
+    * **reason**: holds error messages if there are any 
+
+#### Default CA certificate
+
+By default, when TLS is enabled and CA certificate is not set, we are going to use system bundled one. We can achive that by scanning in `kuma-dp` [default paths](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#example-configuration) for each environment and later send that information in the bootstrap request to the control-plane for a configuration generation.
+
+#### Policy matching
+
+`MeshExternalServices` should work the same with policies as `MeshService`. We are going to introduce a new `kind: MeshExternalService`, which allows targeting them with policies.
```

Comment:
```suggestion
`MeshExternalServices` should work the same with policies as `MeshService`. We are going to introduce a new `kind: MeshExternalService`, which allows targeting them with policies.

We will create separate listeners for each MeshExternalService that bind to VIP provided by HostnameGenerator. It's easier to do it this way instead of having multiple filter chain matches on `passthrough` listener because we this piece of code already implemented and we don't have to search deep to find which FilterChain is responsible for which External Service.
```